contracts/strategy/BnbxYieldConverterStrategy.sol

```diff
@@ -0,0 +1,245 @@
+
+    IERC20Upgradeable private _bnbxToken;
+    IStakeManager private _stakeManager;
+    IMasterVault private _vault;
+
+    struct UserWithdrawRequest {
+        address recipient;
+        uint256 amount;
+        uint256 bnbxAmount;
+        uint256 triggerTime;
+    }
+    mapping(uint256 => UserWithdrawRequest) private _withdrawRequests;
+    uint256 private _firstWithdrawIdx;
+    uint256 private _nextWithdrawIdx;
+
+    uint256 private _bnbxHoldingBalance; // amount of bnbx held by this strategy
+    uint256 private _bnbDepositBalance; // amount of bnb deposited by this strategy
+    uint256 private _bnbxToUnstake; // amount of bnbx to withdraw from stader in next batchWithdraw
+    uint256 private _bnbToDistribute; // amount of bnb to distribute to users who unstaked
+
+    event StakeManagerChanged(address stakeManager);
+
+    /// @dev initialize function - Constructor for Upgradable contract, can be only called once during deployment
+    /// @param destination Address of the stakeManager contract
+    /// @param feeRecipient Address of the fee recipient, which receives yield
+    /// @param bnbxToken Address of BNBx token
+    /// @param masterVault Address of the masterVault contract
+    /// @param stakeManager Address of stakeManager contract
+    function initialize(
+        address destination,
+        address feeRecipient,
+        address bnbxToken,
+        address masterVault,
+        address stakeManager
+    ) public initializer {
+        __BaseStrategy_init(destination, feeRecipient);
+
+        _bnbxToken = IERC20Upgradeable(bnbxToken);
+        _stakeManager = IStakeManager(stakeManager);
+        _vault = IMasterVault(masterVault);
+
+        _bnbxToken.approve(stakeManager, type(uint256).max);
+    }
+
+    /**
+     * Modifiers
+     */
+    modifier onlyVault() {
+        require(msg.sender == address(_vault), "!vault");
+        _;
+    }
+
+    /// @dev deposits the given amount of BNB into Stader stakeManager through bnbxRouter
+    function deposit() external payable onlyVault returns (uint256) {
+        uint256 amount = msg.value;
+        require(amount <= address(this).balance, "insufficient balance");
+
+        return _deposit(amount);
+    }
+
+    /// @dev deposits all the available BNB into Stader stakeManager through bnbxRouter
+    function depositAll() external payable onlyVault returns (uint256) {
+        return _deposit(address(this).balance);
+    }
+
+    /// @dev internal function to deposit the given amount of BNB into Stader stakeManager through bnbxRouter
+    /// @param amount amount of BNB to deposit
+    function _deposit(uint256 amount)
+        internal
+        whenDepositNotPaused
+        returns (uint256)
+    {
+        require(canDeposit(amount), "invalid amount");
+
+        _bnbDepositBalance += amount;
+        _bnbxHoldingBalance += _stakeManager.convertBnbToBnbX(amount);
+        _stakeManager.deposit{value: amount}();
+        return amount;
```

Comment:
If we are not planning to return BNBx to user or master strategy, can we add that to documentation? If we are, we have to transfer it to msg.sender, right?

Reply:
This return just doesn't transfer any token.
It just returns a number `amount of bnb deposited`

method interface
`function deposit() payable external returns(uint256);`

Reply:
sending BNBx to user is not a requirement. Discussed with Dheeraj.